core/headerchain.go

```diff
@@ -308,7 +335,8 @@ func (hc *HeaderChain) GetBlockHashesFromHash(hash common.Hash, max uint64) []co
 	chain := make([]common.Hash, 0, max)
 	for i := uint64(0); i < max; i++ {
 		next := header.ParentHash
-		if header = hc.GetHeader(next); header == nil {
+		nextNum := header.Number.Uint64() - 1
+		if header = hc.getHeader(next, nextNum); header == nil {
```

Comment:
There's not much need to separate out `nextNum`, you can just write 

```
if header = hc.getHeader(next, header.Number.Uint64() - 1); header == nil {
```

next was reused later too but I think there's no need for the extra var for nextNum.